src/webdav/PropertySheets.py

```diff
@@ -0,0 +1,156 @@
+##############################################################################
+#
+# Copyright (c) 2002 Zope Foundation and Contributors.
+#
+# This software is subject to the provisions of the Zope Public License,
+# Version 2.1 (ZPL).  A copy of the ZPL should accompany this distribution.
+# THIS SOFTWARE IS PROVIDED "AS IS" AND ANY AND ALL EXPRESS OR IMPLIED
+# WARRANTIES ARE DISCLAIMED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF TITLE, MERCHANTABILITY, AGAINST INFRINGEMENT, AND FITNESS
+# FOR A PARTICULAR PURPOSE.
+#
+##############################################################################
+
+from html import escape
+
+from AccessControl.class_init import InitializeClass
+from AccessControl.SecurityManagement import getSecurityManager
+from App.Common import iso8601_date
+from App.Common import rfc1123_date
+from OFS.interfaces import IWriteLock
+from OFS.PropertySheets import PropertySheet
+from OFS.PropertySheets import View
+from OFS.PropertySheets import Virtual
+from webdav.common import isDavCollection
+from webdav.common import urlbase
+
+
+def absattr(attr):
```

Comment:
There is another identical implementation in `webdav/common.py`.

Reply:
You're right, I deleted this one.